halpybot/commands/caseutils.py

```diff
@@ -58,41 +97,555 @@ async def cmd_go(ctx: Context, args: List[str]):
     caseboard = ctx.bot.board.by_id
     if not caseboard:
         return await ctx.redirect("The case board is empty!")
-    message = "Here's the current case board:\n"
-    for case in caseboard.values():
-        hskf = "Seal" if case.hull_percent else "Fisher" if case.planet else "Unknown"
+    # Process Args (If Exist)
+    if args:
+        platforms = [
+            member.name.casefold().replace("_horizons", "") for member in Platform
+        ]
+        casetypes = [member.name.casefold() for member in CaseType]
+        list_filter = args[0].casefold()
+        if list_filter not in casetypes and list_filter not in platforms:
+            list_filter = None
+    else:
+        list_filter = None
+
+    filtered_cases = [
+        case
+        for case in caseboard.values()
+        if (not list_filter)
+        or (
+            list_filter
+            in (
+                case.platform.name.casefold().replace("_horizons", ""),
+                case.case_type.name.casefold(),
+            )
+        )
+    ]
+
+    message = "Here's the current case board:\n\n"
+    for case in filtered_cases:
+        hskf = (
+            "Seal"
+            if case.case_type in (CaseType.SEAL, CaseType.BLUE, CaseType.BLACK)
+            else "Fisher"
+            if case.case_type == CaseType.FISH
+            else "Unknown"
+        )
         long_ago = now(tz="utc").diff(case.updated_time).in_words()
         plt = case.platform.name.replace("_", " ")
         message += (
             f"Case {case.board_id}: Client: {case.client_name}, Platform: {plt}, "
             f"Type: {hskf}, Status: {case.status.name}, Updated: {long_ago} ago.\n"
         )
-    return await ctx.redirect(f"{message}{len(caseboard)} Cases on the Board.")
+
+    message += f"\n{len(caseboard)} Cases on the Board."
+    if list_filter:
+        message += (
+            f" Showing {len(filtered_cases)} that match(es) the filter {list_filter!r}."
+        )
+    return await ctx.redirect(message)
 
 
 @Commands.command("listcase")
-async def cmd_listcase(ctx: Context, args: List[str]):
+@Require.permission(Drilled)
+@CaseUtils.gather_case(1)
+async def cmd_listcase(ctx: Context, args: List[str], case: Case):
     """
     Send a user the key details of a case on the board in DMs
 
     Usage: !listcase [board ID]
     Aliases: n/a
     """
-    if not args:
-        return await ctx.reply(get_help_text(ctx.bot.commandsfile, "listcase"))
+    return await ctx.redirect(f"{case}")
+
+
+# CASE MANAGEMENT
+@Commands.command("rename")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_renamecase(ctx: Context, args: List[str], case: Case):
+    """
+    Rename the user of an active case
+
+    Usage: !rename [board ID] [new name]
+    Aliases: n/a
+    """
+    try:
+        await ctx.bot.board.rename_case(args[1], case, ctx.sender)
+    except (AssertionError, ValueError) as err:
+        return await ctx.reply(str(err))
```

Comment:
[error] : casting exceptions to strings destroys 90% of the useful information
[error]: not logging the exception promotes the above warning to an error by severity.

Reply:
Added a logging line in [d0bd809](https://github.com/HullSeals/HalpyBOT/pull/338/commits/d0bd809d1c64a38cf80c32c180ae6b8f5fbddd7e) of 338